multi-int-store/src/test/java/com/connexta/multiintstore/SolrTests.java

```diff
@@ -326,13 +335,63 @@ public String getFilename() {
   }
 
   @Test
-  @Ignore("TODO verify id matches something in S3 before storing to solr")
-  public void testStoreMetadataProductIdNotFound() {
-    // TODO verify 404? 400?
+  public void testStoreMetadataProductIdNotFound() throws Exception {
+    when(mockAmazonS3.doesObjectExist(anyString(), anyString())).thenReturn(Boolean.TRUE);
+
+    // given
+    final URI productLocation = new URI("http://localhost:8080/mis/product/12341");
+    final String queryKeyword = "Winterfell";
+    final String metadataContents =
+        "All the color had been leached from "
+            + queryKeyword
+            + " until only grey and white remained";
+    final String metadataEncoding = "UTF-8";
+    final InputStream metadataInputStream =
+        IOUtils.toInputStream(metadataContents, metadataEncoding);
+    final long metadataFileSize = (long) metadataInputStream.available();
+    final String metadataFileName = "test_file_name.txt";
+    final String metadataMimeType = "text/plain";
+    final MultiValueMap<String, Object> storeMetadataRequestBody = new LinkedMultiValueMap<>();
+    storeMetadataRequestBody.add("fileSize", String.valueOf(metadataFileSize));
+    storeMetadataRequestBody.add("mimeType", metadataMimeType);
+    storeMetadataRequestBody.add(
+        "file",
+        new InputStreamResource(metadataInputStream) {
```

Comment:
:bulb: We override the contentLength and getFilename methods of InputStreamResource many times in anonymous classes. 

![image](https://user-images.githubusercontent.com/133284/62301973-635c4f00-b42e-11e9-98bb-3fc9e51ce554.png)

Let's create a subclass like:

```
public class KnownInputStreamResource extends InputStreamResource {

    // Pass values into constructor
    public KnownInputStreamResource(InputStream istream, long length, String filename) 


    //Override contentLength() and getFilename()

}
```

 Use this class in place of InputStreamResource

```suggestion
        new KnownInputStreamResource(metadataInputStream, metadataFileSize, metadataFileName);
```

Reply:
Good suggestion, I think we should make this a separate PR.